### ISSUE TITLE
Add disjointWith parsing

### ISSUE THREAD

**Issue by minottic** [ExPaNDS-eu/pan-ontologies-api#79]
[OWL file](https://github.com/ExPaNDS-eu/ExPaNDS-experimental-techniques-ontology/commit/c4a22cdc0759dfbd69c68b6a938a6b4671486775#diff-0fc16985c19ffb85fa1e6929b977a44eea0faa168e8bbb9ef6dae946744e95eaR784) now has [disjointWith](https://www.w3.org/TR/owl-ref/#DescriptionAxiom) keyword which requires parsing

**Comment by paulmillar** [comment]
I don't know if this helps, but with [PaNET commit dfce0a0d9d](https://github.com/ExPaNDS-eu/ExPaNDS-experimental-techniques-ontology/commit/dfce0a0d9de7dc08ba9d8cfe1bb0f231ed6e5e78),  we are now generated both the old-style output and the output having run it through a reasoner.

You should find this "reasoned output" as an artefact from the CI/CD pipelines: they are kept for some 7 days.

The reasoned output isn't part of the v1.1.0 release asserts, but we should include it as part of the next release's set of assets.

**Comment by paulmillar** [comment]
Perhaps the easiest way to get the reasoner output is to use the CI/CD-generated artefact.

The general procedure is:
  1. go to [PaNET main page](https://github.com/ExPaNDS-eu/ExPaNDS-experimental-techniques-ontology/)
  2. Click on the green tick near the top of the page, showing the CI/CD result of the most recent commit.
  3. Against the "panet-build / build (push)" row, select "Details" link.
  4. Select "<House> Summary" link (from left-hand side navigation bar).
  5. Under "Artifacts", select the "PaNET" artefact.  This will trigger the download of the `PaNET.zip` file.
  6. Within `PaNET.zip` file, the file `PaNET_reasoned.owl` contains the reasoned output.

For the most recent commit, here is the [PaNET.zip file](https://github.com/ExPaNDS-eu/ExPaNDS-experimental-techniques-ontology/actions/runs/10157922609/artifacts/1754248577).  However, this link will likely break, soon.

**Comment by minottic** [comment]
mmm the reasoned looks very similar to the plain .owl, what's the main advantage to use that? So, apart from downloading the owl and doing the diffs by hand, you don't plan to keep the commited versions of the owl file, apart from attaching to releases?

**Comment by minottic** [comment]
so the reasoner also has the grandchildren computed and referenced as subclassOf?

**Comment by paulmillar** [comment]
> so the reasoned also has the grandchildren computed and annotated with subclassOf grandparent?

Exactly.

This is the RDFS entailment: `subClassOf` is transitive and the reasoner provides the transitive closure of these relationships.

The next PaNET version should provide both a "concise" and a "reasoned" version of PaNET.  For some software (like ontologies-api) there's currently no big difference.  This is because we're largely limited to RDFS entailment and (IIRC) ontologies-api implements support for RDFS entailment.

However, in the future, we might use more OWL features (moving away from only using RDFS relationships) to express more complex relationships in a manageable way.  When doing this, the distinction between the "concise" and "reasoned" versions will likely matter to ontologies-api.

**Comment by minottic** [comment]
ah thanks! So that might be quite useful indeed, I have been thinking that pan-api should rely on existing reasoners rather than parsing the owl itself. Would it make sense to make the `...reasoned.owl` file part of the release assets?

**Comment by paulmillar** [comment]
> pan-api should rely on existing reasoners

Yes, I think this would be a good idea.  But, it's easy for me to say this, when I don't have to do the work!

>  Would it make sense to make the ...reasoned.owl file part of the release assets?

Yes, I would say so.  The the primary reason for updating the build container (to run the result through a reasoner and save the result) was to include the reasoned output as a release asset.

I'd say that, assuming it isn't urgent, we can do this as part of the next PaNET release.
